src/DotNetLightning.Core/Channel/CommitmentsModule.fs

```diff
@@ -415,3 +416,80 @@ module internal Commitments =
+        do! check (transaction.Version = TxVersionNumberOfCommitmentTxs)
+        let! txIn = Seq.tryExactlyOne transaction.Inputs
+        do! check (fundingOutPoint = txIn.PrevOut)
+        let! obscuredCommitmentNumber =
+            ObscuredCommitmentNumber.TryFromLockTimeAndSequence transaction.LockTime txIn.Sequence
+        return obscuredCommitmentNumber
+    }
+
+    let tryGetFundsFromLocalCommitmentTx (commitments: Commitments)
+                                         (localChannelPrivKeys: ChannelPrivKeys)
+                                         (network: Network)
+                                         (transaction: Transaction)
+                                             : Option<TransactionBuilder> = option {
+        let! obscuredCommitmentNumber =
+            tryGetObscuredCommitmentNumber
+                commitments.FundingScriptCoin.Outpoint
+                transaction
+        let localChannelPubKeys = commitments.LocalParams.ChannelPubKeys
+        let remoteChannelPubKeys = commitments.RemoteParams.ChannelPubKeys
+        let commitmentNumber =
+            obscuredCommitmentNumber.Unobscure
+                commitments.LocalParams.IsFunder
+                localChannelPubKeys.PaymentBasepoint
+                remoteChannelPubKeys.PaymentBasepoint
+
+        let perCommitmentPoint =
+            localChannelPrivKeys.CommitmentSeed.DerivePerCommitmentPoint commitmentNumber
+        let localCommitmentPubKeys =
+            perCommitmentPoint.DeriveCommitmentPubKeys localChannelPubKeys
+        let remoteCommitmentPubKeys =
+            perCommitmentPoint.DeriveCommitmentPubKeys remoteChannelPubKeys
+
+        let transactionBuilder = network.CreateTransactionBuilder()
+
+        let toLocalScriptPubKey =
+            Scripts.toLocalDelayed
+                remoteCommitmentPubKeys.RevocationPubKey
+                commitments.LocalParams.ToSelfDelay
+                localCommitmentPubKeys.DelayedPaymentPubKey
+        let! toLocalIndex =
+            let toLocalWitScriptPubKey = toLocalScriptPubKey.WitHash.ScriptPubKey
+            Seq.tryFindIndex
+                (fun (txOut: TxOut) -> txOut.ScriptPubKey = toLocalWitScriptPubKey)
+                transaction.Outputs
+
+        let delayedPaymentPrivKey =
+            perCommitmentPoint.DeriveDelayedPaymentPrivKey
+                localChannelPrivKeys.DelayedPaymentBasepointSecret
+
+        transactionBuilder.SetVersion TxVersionNumberOfCommitmentTxs
+        |> ignore
+        transactionBuilder.Extensions.Add (CommitmentToLocalExtension())
+        transactionBuilder.AddKeys (delayedPaymentPrivKey.RawKey()) |> ignore
+        transactionBuilder.AddCoin(
+            ScriptCoin(transaction, uint32 toLocalIndex, toLocalScriptPubKey),
+            CoinOptions(
+                Sequence = (Nullable <| Sequence(uint32 commitments.LocalParams.ToSelfDelay.Value))
+            )
+        ) |> ignore
+
+        return transactionBuilder
```

Comment:
```suggestion
        return
            transactionBuilder
                .AddKeys(delayedPaymentPrivKey.RawKey())
                .AddCoin(
                    ScriptCoin(transaction, uint32 toLocalIndex, toLocalScriptPubKey), CoinOptions(
                        Sequence = (Nullable <| Sequence(uint32 commitments.LocalParams.ToSelfDelay.Value))
                    )
            )
```

Reply:
*uugh..* Your code works, however my code didn't. The difference is that I had a space between `AddKeys` and the open-paren:

```
.AddKeys (delayedPaymentPrivKey.RawKey())
        ^ here
```

Reply:
Fixed.

Reply:
I was also surprised when I discovered this interesting behaviour from the F# compiler... for more info see this thread: https://github.com/fsprojects/fantomas/issues/1437